client/cmd/resource/change_namespace.go

```diff
@@ -0,0 +1,188 @@
+}
+
+func (c *changeNamespaceCommand) PreRunE(_ *cobra.Command, _ []string) error {
+	// Load mandatory config
+	conf, err := config.LoadClientConfig(c.configFilePath)
+	if err != nil {
+		return err
+	}
+
+	c.clientConfig = conf
+	return err
+}
+
+func (c *changeNamespaceCommand) RunE(_ *cobra.Command, args []string) error {
+	resourceFullName := args[0]
+	c.dataStore = args[1]
+	err := c.sendChangeNamespaceRequest(resourceFullName)
+	if err != nil {
+		return fmt.Errorf("namespace change request failed for resource %s: %w", resourceFullName, err)
+	}
+	c.logger.Info("successfully changed namespace and deployed new DAG on Scheduler")
+	return nil
+}
+
+func (c *changeNamespaceCommand) sendChangeNamespaceRequest(resourceName string) error {
+	conn, err := connectivity.NewConnectivity(c.host, changeNamespaceTimeout)
+	if err != nil {
+		return err
+	}
+	defer conn.Close()
+
+	// fetch Instance by calling the optimus API
+	resourceRunServiceClient := pb.NewResourceServiceClient(conn.GetConnection())
+	request := &pb.ChangeResourceNamespaceRequest{
+		ProjectName:      c.project,
+		NamespaceName:    c.oldNamespace,
+		DatastoreName:    c.dataStore,
+		ResourceName:     resourceName,
+		NewNamespaceName: c.newNamespace,
+	}
+
+	_, err = resourceRunServiceClient.ChangeResourceNamespace(conn.GetContext(), request)
+	return err
+}
+
+func (c *changeNamespaceCommand) PostRunE(_ *cobra.Command, args []string) error {
+	c.logger.Info("\n[INFO] Moving resource in filesystem")
+	resourceName := args[0]
+	c.dataStore = args[1]
+
+	oldNamespaceConfig, err := c.getResourceDatastoreConfig(c.oldNamespace, c.dataStore)
+	if err != nil {
+		return errors.Wrap(tenant.EntityNamespace, "unregistered old namespace", err)
+	}
+
+	resourceSpecReadWriter, err := specio.NewResourceSpecReadWriter(afero.NewOsFs())
+	if err != nil {
+		return err
+	}
+
+	resourceSpec, err := resourceSpecReadWriter.ReadByName(oldNamespaceConfig.Path, resourceName)
+	if err != nil {
+		return errors.Wrap(tenant.EntityNamespace, "unable to find resource in old namespace", err)
+	}
+
+	fs := afero.NewOsFs()
+	newNamespaceConfig, err := c.getResourceDatastoreConfig(c.newNamespace, c.dataStore)
+	if err != nil || newNamespaceConfig.Path == "" {
+		c.logger.Warn("[warn] new namespace not recognised for Resources")
+		c.logger.Warn("[info] removing resource from old namespace")
+		err = fs.RemoveAll(resourceSpec.Path)
+		if err != nil {
+			c.logger.Warn("unable to remove resource from old namespace")
+			return errors.NewError(errors.ErrInternalError, "change-namespace", "unable to remove resource from old namespace")
+		}
+		c.logger.Warn("[info] removed resource spec from current namespace directory")
+		c.logger.Warn("[info] run `optimus resource export` on the new namespace repo, to fetch the newly moved resource.")
+		c.logger.Info("[OK] Resource moved successfully")
+		return nil
+	}
+
+	var relativeResourcePath string
+	splitComp := strings.Split(resourceSpec.Path, oldNamespaceConfig.Path)
```

Comment:
`newResourcePath = strings.Replace(resourceSpec.Path, oldNamespaceConfig.Path, newNamespaceConfig.Path, 1)`, we can use this and remove line 154-159

Reply:
nice idea

Reply:
done